docs-src/concepts/what-is-versioning.md

```diff
@@ -0,0 +1,170 @@
+
+- **Increased operational overhead**: There could be increased operational overhead in managing different versions of Workers and ensuring compatibility, particularly in larger systems with many Workers.
+
+### What is a Build ID?
+
+A Build ID is any string that you associate with a version of a Workflow. As a user-generated string, it's used to track, manage, and reference different stages or versions of the Workflow.
+
+In the context of a Workflow with Version Sets, a Build ID can be assigned as the default for a Version Set or as the overall default for the entire Workflow.
+
+A Build ID can contain a maximum of 255 characters.
+
+#### What is a default Build ID?
+
+A Default Build ID is the specific Build ID that a Worker refers to when executing Tasks.
+While this is typically the most recent Build ID, you can manually specify a different Build ID as the default when registering the Workflow with a Worker.
+This option is beneficial for situations where you prefer an older Build ID to be the default rather than the latest one.
+A unique Default Build ID can be assigned to both the Task Queue and each individual Compatible Version Set.
+
+If you have one Build ID, it's known as the Default Build ID.
+
+### What is a Version Set?
+
+A Version Set groups multiple Build IDs for compatibility, management, and display purposes within a Workflow.
+Each Version Set contains a designated Default Build ID, and the entire Workflow has one overall Default Build ID.
+
+#### What is a Compatible Version Set?
+
+A Compatible Version Set is a user-defined collection of Build IDs that indicates which versions of the Workflow code are mutually compatible.
+Each Compatible Version Set has a Default Build ID.
+
+A Task Queue has a default maximum of 10 Compatible Version Sets and a maximum of 100 Build IDs for a single Task Queue across all Compatible Version Sets.
+
+You can visualize Compatible Version Sets as a tree-like structure, where a branch represents the progressive evolution of each build. Each branch in the tree is a series of versions that are compatible with each other, but not necessarily with versions in other branches.
+
+For example, Build ID of `1.0` might start a new branch and therefore isn't compatible with Build ID of `0.4` from an earlier branch. However, Build ID of `1.1` is compatible with Build ID of `1.0` because it's a newer version of the same branch.
+
+## Task Queue-based Versioning
+
+While Task Queue-based Versioning is available to use, it's recommended to use the [Worker Versioning](#worker-versioning) approach instead, as it simplifies the process of versioning your Task Queues.
+
+Task Queue-based Versioning is the process of creating a new Task Queue for each version of your Workflow, and then modify the Task Queue the Client points to.
+
+Workers pull from Task Queues, and because of that you start Workflows on Task Queues, and each Task Queues have a unique identifier.
```

Comment:
```suggestion
Workers pull from Task Queues, and because of that you start Workflows on Task Queues, and each Task Queue has a unique identifier.
```